java/java.hints/src/org/netbeans/modules/java/hints/errors/MagicSurroundWithTryCatchFix.java

```diff
@@ -396,36 +417,65 @@ private static StatementTree createLogStatement(CompilationInfo info, TreeMaker
+                    // and "contentEquals" fails, so HACK! use litteral class name.
+                    if (type.getKind() == TypeKind.DECLARED && ((TypeElement)((DeclaredType)type).asElement()).getQualifiedName().contentEquals(loggerFQN)) {
+                        existingLogger = ve;
+                    }
+                }
+            } catch(IOException ex) {
+            }
+        }
+
         // finally, make the invocation
-        ExpressionTree etExpression = make.MethodInvocation(
-                Collections.<ExpressionTree>emptyList(),
-                make.MemberSelect(
-                useFQN ? make.Identifier(logger.toString()) : make.QualIdent(logger),
-                "getLogger"),
-                Collections.<ExpressionTree>singletonList(arg));
-        ExpressionTree levelExpression = make.MemberSelect(make.QualIdent(level), "SEVERE");
-
-        return make.ExpressionStatement(make.MethodInvocation(Collections.<ExpressionTree>emptyList(), make.MemberSelect(etExpression, "log"), Arrays.asList(levelExpression, make.Literal(null), make.Identifier(name))));
-    }
+        ExpressionTree etExpression;
+        if (existingLogger != null) {
+            etExpression = make.Identifier(existingLogger);
+        } else {
+            etExpression = make.MethodInvocation(
+                    Collections.<ExpressionTree>emptyList(),
+                    make.MemberSelect(
+                        useFQN ? make.Identifier(loggerFactory.toString()) : make.QualIdent(loggerFactory),
+                        "getLogger"),
+                    Collections.<ExpressionTree>singletonList(arg));
+        }
+        ExpressionTree levelExpression = make.MemberSelect(make.QualIdent(level), levelName);
+        ExpressionTree nullMsgArg = isSystemLogger
+                ? (ExpressionTree)org.netbeans.modules.java.hints.spiimpl.Utilities.parseAndAttribute(info, "(String)null", null)
+                : make.Literal(null);
         
+        return make.ExpressionStatement(make.MethodInvocation(Collections.<ExpressionTree>emptyList(), make.MemberSelect(etExpression, "log"), Arrays.asList(levelExpression, nullMsgArg, make.Identifier(name))));
+    }
+    
     private static StatementTree createRethrowAsRuntimeExceptionStatement(WorkingCopy info, TreeMaker make, String name) {
         if (!ErrorFixesFakeHint.isRethrowAsRuntimeException(ErrorFixesFakeHint.getPreferences(info.getFileObject(), FixKind.SURROUND_WITH_TRY_CATCH))) {
             return null;
         }
-
+        
         TypeElement runtimeException = info.getElements().getTypeElement("java.lang.RuntimeException");
-
+        
         if (runtimeException == null) {
             return null;
         }
-
+        
         ExpressionTree exceptionName = make.QualIdent(runtimeException);
         StatementTree result = make.Throw(make.NewClass(null, Collections.<ExpressionTree>emptyList(), exceptionName, Arrays.asList(make.Identifier(name)), null));
-
+        
```

Comment:
Nitpick: New whitespace introduced.

Reply:
That's weird. Code I didn't modify. Wonder why NB decided to do that. I notice that there are scattered places around the code where there are blank lines with spaces in them.

Fixing.